### PR TITLE
build: update to rxjs 7

### LINKED ISSUE #24371: Update rxjs to v7.x
### Command

new

### Description

Update rxjs to v7.x

### Describe the solution you'd like

rxjs is a quite heavy package at least in terms of the number of files ([v6.x ~3.7K, v7.x ~2.3K](https://packagephobia.com/result?p=rxjs%406.6.7%2Crxjs%407.6.0)). It'd be nice if the `@angular-devkit` packages updated to rxjs v7.x so that it can be deduped by package managers. An example from [keycloak-angular](https://github.com/mauriciovigolo/keycloak-angular):

```
C:\Users\xmr\Desktop\keycloak-angular>npm ls rxjs
keycloak-angular@13.0.0 C:\Users\xmr\Desktop\keycloak-angular
+-- @angular-devkit/build-angular@15.0.2
| +-- @angular-devkit/architect@0.1500.2
| | `-- rxjs@6.6.7
| +-- @angular-devkit/build-webpack@0.1500.2
| | `-- rxjs@6.6.7
| +-- inquirer@8.2.4
| | `-- rxjs@7.6.0 deduped
| `-- rxjs@6.6.7
+-- @angular-devkit/core@15.0.2
| `-- rxjs@6.6.7
+-- @angular/cli@15.0.2
| `-- @angular-devkit/schematics@15.0.2
|   `-- rxjs@6.6.7
+-- @angular/common@15.0.2
| `-- rxjs@7.6.0 deduped
+-- @angular/core@15.0.2
| `-- rxjs@7.6.0 deduped
+-- @angular/router@15.0.2
| `-- rxjs@7.6.0 deduped
+-- concurrently@7.6.0
| `-- rxjs@7.6.0 deduped
+-- ng-packagr@15.0.1
| `-- rxjs@7.6.0 deduped
`-- rxjs@7.6.0
```

Diff if I remove the duplicate rxjs 6.x packages from `node_modules`:

```
327 MB (343.012.651 bytes)
50.909 Files, 6.350 Folders

302 MB (317.352.111 bytes)
32.816 Files, 5.910 Folders
```

As you can see, even with fast disks and fast connections, this would significantly reduce the disk churn.

---

Sorry if there's had been an issue about it already, I searched but I couldn't find an existent issue.

### Describe alternatives you've considered

_No response_

### PR DESCRIPTION
**Pull request by alan-agius4**
G3 is now using RXJS version 7 which makes it possible for the CLI to also be updated to RXJS 7.

NB: this change does not remove all usages of the deprecated APIs.

Closes #24371

### CODE REVIEW

File: packages/angular_devkit/architect/src/api.ts [hunk 1]
```diff
@@ -77,6 +77,12 @@ export interface BuilderRun {
    */
   result: Promise<BuilderOutput>;
 
+  /**
+   * The lasy output from a builder. This is recommended when scheduling a builder and only being
```
Comment (clydin) [inline packages/angular_devkit/architect/src/api.ts:81]:
```suggestion
   * The last output from a builder. This is recommended when scheduling a builder and only being
```

File: packages/angular_devkit/architect/src/schedule-by-name.ts [hunk 2]
```diff
@@ -101,29 +98,26 @@ export async function scheduleByName(
...
+    async stop() {
       job.stop();
-
-      return job.outboundBus
-        .pipe(
-          ignoreElements(),
-          catchError(() => EMPTY),
-        )
-        .toPromise();
+      outboundBusSub.unsubscribe();
```
Comment (clydin) [inline packages/angular_devkit/architect/src/schedule-by-name.ts:120]: Does this result in any different behavior?
Looks like it was previously waiting for the outbound bus to be complete before resolving.

Comment (alan-agius4) [reply packages/angular_devkit/architect/src/schedule-by-name.ts:120]: I did not see any different behaviour. But, I can revert this.

**Comment by alan-agius4** [comment]
@clydin PTAL

**Review by clydin — APPROVED** [review]

**Comment by HyperLife1119** [comment]
Should the `'rxjs/operators'` import point also be replaced with `'rxjs'`?
https://rxjs.dev/guide/importing#new-in-rxjs-v720

**Pull request merged**